main/src/main/scala/sbt/Keys.scala

```diff
@@ -399,10 +399,11 @@ object Keys {
   val bspConfig = taskKey[Unit]("Create or update the BSP connection files").withRank(DSetting)
   val bspEnabled = SettingKey[Boolean](BasicKeys.bspEnabled)
   val bspTargetIdentifier = settingKey[BuildTargetIdentifier]("Build target identifier of a project and configuration.").withRank(DSetting)
-  val bspWorkspace = settingKey[Map[BuildTargetIdentifier, Scope]]("Mapping of BSP build targets to sbt scopes").withRank(DSetting)
+  val bspWorkspace = settingKey[BspWorkspace]("Mapping of BSP build targets to sbt scopes").withRank(DSetting)
```

Comment:
In theory that will break the compatibility between sbt versions. In that particular case, I don't think any build or plugin depends on it directly. So it should not be a problem. But maybe we should take the opportunity to make it private?

Reply:
It's not strictly necessary to change this key but it sure seemed the cleanest while prototyping. Happy to take direction though about how to proceed to reduce risk of compatibility breakage.

Reply:
It's used by at least one plugin - https://github.com/scalacenter/scala-debug-adapter/blob/b3125d8a3873a7218e452b256ed938d14c401c5c/sbt/plugin/src/main/scala/ch/epfl/scala/debugadapter/sbtplugin/DebugAdapterPlugin.scala#L68, but I am guessing we can make exception for that one.

Reply:
That's right... How did I forget about it? So `bspWorkspace` cannot be private. Also if the signature changes it will be tricky for us, but we can probably work it out anyway.

Reply:
I'll rework to avoid a signature change. I can define a new key that aggregates the `Scope` and `LoadedBuildUnit` based targets and leave `val bspWorkspace = settingKey[Map[BuildTargetIdentifier, Scope]]` as a view on the scope part of that data for backwards compatibility.